.github/workflows/on-merge.yml

```diff
@@ -5,28 +5,36 @@ on:
       - labeled
       - unlabeled
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
 jobs:
   on-merge:
     name: 'Label and Backport'
     runs-on: ubuntu-latest
     if: |
-      github.event.pull_request.merged == true &&
-        (github.event.action == 'closed' ||
-          (github.event.action == 'labeled' &&
-            (github.event.label.name == 'backport:prev-minor' ||
-              github.event.label.name == 'backport:prev-major' ||
-              github.event.label.name == 'backport:current-major' ||
-              github.event.label.name == 'backport:all-open' ||
-              github.event.label.name == 'backport:version' ||
-              github.event.label.name == 'auto-backport')) ||
-          (github.event.action == 'unlabeled' &&
-            github.event.label.name == 'backport:skip' &&
-            (contains(github.event.pull_request.labels.*.name, 'backport:prev-minor') ||
-              contains(github.event.pull_request.labels.*.name, 'backport:prev-major') ||
-              contains(github.event.pull_request.labels.*.name, 'backport:current-major') ||
-              contains(github.event.pull_request.labels.*.name, 'backport:all-open') ||
-              contains(github.event.pull_request.labels.*.name, 'backport:version') ||
-              contains(github.event.pull_request.labels.*.name, 'auto-backport'))))
+      github.event.pull_request.merged == true && (
+        (
+          github.event.action == 'closed'
+        ) || (
+          github.event.action == 'labeled' && (
+            github.event.label.name == 'backport:all-open' ||
+            github.event.label.name == 'backport:version'
+          )
+        ) || (
+          github.event.action == 'labeled' && contains(github.event.pull_request.labels.*.name, 'backport:version') && (
+            startsWith(github.event.label.name,'v7') ||
```

Comment:
not sure if its worth it to improve this further as a next item, but maybe we can run a pre action, read the versions file, populate the available branch versions as a VAR instead of having those hardcoded. That way is one less thing to worry about . In any case its low hanging fruit I would say since major branches are not expected to change much

Reply:
Check in 2cf4584a255afe38518e06cdb69f4ac46889af29